src/features/vault.ts

```diff
@@ -0,0 +1,241 @@
+      console.log('Encrypt selected text');
+      const vaultId: string | undefined = await askForVaultId(keyInCfg);
+      if (!vaultId) {
+        displayMissingIdentityError();
+        return;
+      }
+
+      const encryptedText = `!vault |\n${encryptInline(
+        text,
+        rootPath,
+        vaultId,
+        config
+      )}`;
+      editor.edit((editBuilder) => {
+        editBuilder.replace(
+          selection,
+          encryptedText.replace(
+            /\n/g,
+            `\n${' '.repeat(selection.start.character)}`
+          )
+        );
+      });
+    } else if (type === 'encrypted') {
+      console.log('Decrypt selected text');
+
+      const decryptedText = await decryptInline(text, rootPath, config);
+      editor.edit((editBuilder) => {
+        editBuilder.replace(selection, decryptedText);
+      });
+    }
+  } else {
+    let content = '';
+    await vscode.workspace.openTextDocument(doc.fileName).then((document) => {
+      content = document.getText();
+    });
+    const type = getTextType(content);
+
+    if (type === 'plaintext') {
+      console.log('Encrypt entire file');
+      const vaultId: string | undefined = await askForVaultId(keyInCfg);
+      if (!vaultId) {
+        displayMissingIdentityError();
+        return;
+      }
+      encryptFile(doc.fileName, rootPath, vaultId, config);
+      vscode.window.showInformationMessage(`File encrypted: '${doc.fileName}'`);
+    } else if (type === 'encrypted') {
+      console.log('Decrypt entire file');
+
+      await decryptFile(doc.fileName, rootPath, config);
+      vscode.window.showInformationMessage(`File decrypted: '${doc.fileName}'`);
+    }
+    vscode.commands.executeCommand('workbench.action.files.revert');
+  }
+};
+
+// Returns whether the selected text is encrypted or in plain text.
+const getInlineTextType = (text: string) => {
+  if (text.trim().startsWith('!vault |')) {
+    text = text.replace('!vault |', '');
+  }
+
+  return text.trim().startsWith('$ANSIBLE_VAULT;') ? 'encrypted' : 'plaintext';
+};
+
+// Returns wheter the file is encrypted or in plain text.
+const getTextType = (text: string) => {
+  return text.indexOf('$ANSIBLE_VAULT;') === 0 ? 'encrypted' : 'plaintext';
+};
+
+const encryptInline = async (
+  text: string,
+  rootPath: string | undefined,
+  vaultId: string,
+  config: vscode.WorkspaceConfiguration
+) => {
+  const tmpFilename = await tmp.tmpName();
+  await fs.promises.writeFile(tmpFilename, Buffer.from(text, 'utf8'));
+  console.log(`Wrote encrypted string to temporary file '${tmpFilename}'`);
+
+  encryptFile(tmpFilename, rootPath, vaultId, config);
+  const encryptedText = await fs.promises.readFile(tmpFilename, 'utf8');
+  console.log(`encryptedText == '${encryptedText}'`);
+
+  if (!!tmpFilename) {
+    await fs.promises.unlink(tmpFilename);
+    console.log(`Removed temporary file: '${tmpFilename}'`);
+  }
+
+  return encryptedText.trim();
+};
+
+const decryptInline = async (
```

Comment:
Oh, I've missed with the previous comments about ACLs, most of them are probably more important wrt this function.

Reply:
I've modified the original code so that it doesn't create any temporary files during inline de- or encryption, it just pipes the text directly through ansible-vault.